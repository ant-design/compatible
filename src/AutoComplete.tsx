import * as React from 'react';
import { AutoComplete } from 'antd';
import warning from 'rc-util/lib/warning';

type AutoCompleteProps = Parameters<typeof AutoComplete>[0];
type AutoCompleteRef = AutoCompleteProps['ref'];

type CompatibleAutoCompleteProps = AutoCompleteProps & {
  /** @deprecated Please use `popupClassName` instead. */
  dropdownClassName?: string;
};

const CompatibleAutoComplete = React.forwardRef(
  (
    { dropdownClassName, popupClassName, ...restProps }: CompatibleAutoCompleteProps,
    ref: AutoCompleteRef,
  ) => {
    warning(
      !dropdownClassName,
      "[antd: AutoComplete] `dropdownClassName` is removed in v5, please use `popupClassName` instead.",
    );

    return (
      <AutoComplete {...restProps} popupClassName={popupClassName || dropdownClassName} ref={ref} />
    );
  },
);

if (process.env.NODE_ENV !== 'production') {
  CompatibleAutoComplete.displayName = 'CompatibleAutoComplete';
}

export default CompatibleAutoComplete;
