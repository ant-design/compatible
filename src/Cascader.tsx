import * as React from 'react';
import { Cascader } from 'antd';
import warning from 'rc-util/lib/warning';

type CascaderProps = Parameters<typeof Cascader>[0];
type CascaderRef = CascaderProps['ref'];

type CompatibleCascaderProps = CascaderProps & {
  /** @deprecated Please use `popupClassName` instead. */
  dropdownClassName?: string;
};

const CompatibleCascader = React.forwardRef(
  (
    { dropdownClassName, popupClassName, ...restProps }: CompatibleCascaderProps,
    ref: CascaderRef,
  ) => {
    warning(
      !dropdownClassName,
      `[Compatible: Cascader] 'dropdownClassName' is deprecated. Please use 'popupClassName' instead.`,
    );

    return (
      <Cascader {...restProps} popupClassName={popupClassName || dropdownClassName} ref={ref} />
    );
  },
);

if (process.env.NODE_ENV !== 'production') {
  CompatibleCascader.displayName = 'CompatibleCascader';
}

export default CompatibleCascader;
