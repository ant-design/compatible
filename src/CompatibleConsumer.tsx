import { ConfigProvider } from 'antd';
import type { ConfigConsumerProps } from 'antd/lib/config-provider';

export type { ConfigConsumerProps };

const MergedConfigConsumer = ConfigProvider.ConfigContext.Consumer;

export default MergedConfigConsumer;
