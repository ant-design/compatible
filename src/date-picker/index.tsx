import * as React from 'react';
import RcCalendar from 'rc-calendar';
import MonthCalendar from 'rc-calendar/lib/MonthCalendar';
import createPicker from './createPicker';
import wrapPicker from './wrapPicker';
import RangePicker from './RangePicker';
import WeekPicker from './WeekPicker';
import { DatePickerProps, DatePickerDecorator } from './interface';

const DatePicker = wrapPicker(
  createPicker(RcCalendar),
  'date',
) as React.ClassicComponentClass<DatePickerProps>;

const MonthPicker = wrapPicker(createPicker(MonthCalendar), 'month');

Object.assign(DatePicker, {
  RangePicker: wrapPicker(RangePicker, 'date'),
  MonthPicker,
  WeekPicker: wrapPicker(WeekPicker, 'week'),
});

export default DatePicker as DatePickerDecorator;
