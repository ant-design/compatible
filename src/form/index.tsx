import Form from './Form';

export {
  FormProps,
  FormComponentProps,
  FormCreateOption,
  ValidateCallback,
  ValidationRule,
} from './Form';
export { FormItemProps } from './FormItem';

export default Form;
