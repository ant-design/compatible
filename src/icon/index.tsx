import React from 'react';
import * as allIcons from '@ant-design/icons/lib/icons';
import AntdIcon, {
  createFromIconfontCN,
  getTwoToneColor,
  setTwoToneColor,
} from '@ant-design/icons';

import {
  withThemeSuffix,
  removeTypeTheme,
  getThemeFromTypeName,
  alias,
} from './utils';
import warning from '../_util/warning';

export interface CustomIconComponentProps {
  width: string | number;
  height: string | number;
  fill: string;
  viewBox?: string;
  className?: string;
  style?: React.CSSProperties;
  spin?: boolean;
  rotate?: number;
  ['aria-hidden']?: React.AriaAttributes['aria-hidden'];
}

export type ThemeType = 'filled' | 'outlined' | 'twoTone';

interface CoreIconProps {
  tabIndex?: number;
  className?: string;
  theme?: ThemeType;
  title?: string;
  onKeyUp?: React.KeyboardEventHandler<HTMLElement>;
  onClick?: React.MouseEventHandler<HTMLElement>;
  twoToneColor?: string;
  viewBox?: string;
  spin?: boolean;
  rotate?: number;
  style?: React.CSSProperties;
  role?: string;
}

interface LegacyTypeIconProps extends CoreIconProps {
  type: string;
}

export interface IconProps extends CoreIconProps {
  type?: string;
  component?: React.ComponentType<CustomIconComponentProps>;
}

export interface IconComponent<P> extends React.SFC<P> {
  createFromIconfontCN: typeof createFromIconfontCN;
  getTwoToneColor: typeof getTwoToneColor;
  setTwoToneColor: typeof setTwoToneColor;
}

const iconsMap: {
  [key: string]: any;
} = allIcons;

const LegacyTypeIcon: React.FC<LegacyTypeIconProps> = props => {
  const { type, theme } = props;

  if (theme) {
    const themeInName = getThemeFromTypeName(type);
    warning(
      !themeInName || theme === themeInName,
      'Icon',
      `The icon name '${type}' already specify a theme '${themeInName}',` +
        ` the 'theme' prop '${theme}' will be ignored.`,
    );
  }
  const computedType = withThemeSuffix(
    removeTypeTheme(alias(type)),
    theme || 'outlined',
  );
  const targetIconComponent = iconsMap[computedType];
  warning(
    targetIconComponent,
    'Icon',
    `The icon name '${type}'${
      theme ? `with ${theme}` : ''
      } doesn't exist, please check it at https://ant.design/components/icon`,
  );

  return targetIconComponent
    ? React.createElement(targetIconComponent, props)
    : null;
};

const Icon: IconComponent<IconProps> = props => {
  const { type, component, children } = props;
  warning(
    Boolean(type || component || children),
    'Icon',
    'Should have `type` prop or `component` prop or `children`.',
  );

  if (component || children) {
    return <AntdIcon {...props} />;
  }

  if (typeof type === 'string') {
    return <LegacyTypeIcon {...props} type={type} />;
  }

  return <AntdIcon />;
};

Icon.createFromIconfontCN = createFromIconfontCN;
Icon.getTwoToneColor = getTwoToneColor;
Icon.setTwoToneColor = setTwoToneColor;

export default Icon;
