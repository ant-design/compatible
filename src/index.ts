export { default as Icon } from './icon';
export { default as Form } from './form';
export { default as Mention } from './mention';
