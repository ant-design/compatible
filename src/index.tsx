export { default as Comment } from './comment';
