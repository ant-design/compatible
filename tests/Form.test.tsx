import React from 'react';
import { ConfigProvider } from 'antd';
import { Form } from '../src';
import { render } from '@testing-library/react';

describe('Form', () => {
  it('no hash', () => {
    const { container } = render(
      <ConfigProvider
        theme={{
          hashed: false,
        }}
      >
        <Form />
      </ConfigProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
