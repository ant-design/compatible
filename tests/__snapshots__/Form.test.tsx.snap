// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form no hash 1`] = `
<div>
  <form
    class="ant-legacy-form ant-legacy-form-horizontal"
  />
</div>
`;
